src/util/websocket/Common.h

```diff
@@ -0,0 +1,108 @@
+//  Copyright 2023, University of Freiburg,
+//  Chair of Algorithms and Data Structures.
+//  Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#pragma once
+
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <random>
+#include <type_traits>
+
+#include "../../engine/RuntimeInformation.h"
+
+namespace ad_utility::websocket::common {
+
+class QueryId {
+  std::string id_;
+  explicit QueryId(std::string id) : id_{std::move(id)} {}
+
+ public:
+  static QueryId idFromString(std::string id) { return QueryId{std::move(id)}; }
+  bool empty() const noexcept { return id_.empty(); }
+
+  // Starting with gcc 12 and clang 15 these can be constexpr
+  bool operator==(const QueryId&) const noexcept = default;
+  // required for multimap
+  auto operator<=>(const QueryId&) const noexcept = default;
+
+  friend std::hash<QueryId>;
+};
+}  // namespace ad_utility::websocket::common
+
+template <>
+struct std::hash<ad_utility::websocket::common::QueryId> {
+  auto operator()(
+      const ad_utility::websocket::common::QueryId& queryId) const noexcept {
+    return std::hash<std::string>{}(queryId.id_);
+  }
+};
+
+namespace ad_utility::websocket::common {
+class OwningQueryId {
```

Comment:
This is a wrapper for `QueryId` which in practice is owned by the query execution context and destroyed along with it